modules/base/shaders/model_fs.glsl

```diff
@@ -165,14 +175,45 @@ Fragment getFragment() {
       vec3 specularColor =
         specularIntensity * lightColor * specularFactor * specularAlbedo;
 
-      totalLightColor += lightIntensities[i] * (diffuseColor + specularColor);
+      totalLightColor += lightIntensities[i] * diffuseColor;
+      totalSpecularColor += lightIntensities[i] * specularColor;
+    }
+    frag.color.rgb = totalLightColor + totalSpecularColor;
+
+    if (has_shadow_depth_map) {
```

Comment:
These calculations are hard to understand when you are not familiar with what is happening. Could we maybe add some comments that explain the formulas and where they are coming from?